production/helm/loki/templates/_helpers.tpl

```diff
@@ -1083,7 +1083,7 @@ enableServiceLinks: false
 {{- end }}
 
 {{- define "loki.config.checksum" -}}
-checksum/config: {{ include (print .Template.BasePath "/config.yaml") . | sha256sum }}
+checksum/config: {{ get (include (print .Template.BasePath "/config.yaml") . | fromYaml) "data" | toYaml | sha256sum }}
```

Comment:
Could this be extracted into a helper function?

Reply:
It could be, yes. But we have a parameterization here, which is the actual config yaml file name, which is not always `/config.yaml`. It may also be `/gateway/configmap-gateway.yaml` in deployment-gateway-nginx.
That means, we likely want to create a "parameter dict(ionary)", something like:
```yaml
{{/*
compute a ConfigMap or Secret checksum only based on the .data content
*/}}
{{- define "loki.configMapOrSecretContentHash" -}}
{{ get (include (print .Template.BasePath .name) .ctx | fromYaml) "data" | toYaml | sha256sum }}
{{- end }}
```
and usage:
```yaml
checksum/config: {{ include "loki.configMapOrSecretContentHash" (dict "ctx" . "name" "/config.yaml") }}
```
and:
```yaml
checksum/config: {{ include "loki.configMapOrSecretContentHash" (dict "ctx" . "name" "/gateway/configmap-gateway.yaml") }}
```
That may look uncommon, but does work.

Reply:
I have implemented it like that now.